clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -780,7 +782,7 @@ public void run() {
         }
         if (pushJobSetting.sendControlMessagesDirectly) {
           getVeniceWriter(pushJobSetting).broadcastStartOfPush(
-              SORTED,
+              !pushJobSetting.isBatchWriteOptimizationForHybridStoreEnabled,
```

Comment:
Shouldn’t we consider both the store type and the config setting when deciding whether to enable the optimization?

```java
// Enable optimized batch ingestion only if the store is batch-only,
// or if the optimization is explicitly enabled for hybrid stores.
boolean enableOptimizedBatchIngestion =
    !store.isHybrid() || pushJobSetting.isBatchWriteOptimizationForHybridStoreEnabled;
```